src/converters.jl

```diff
@@ -0,0 +1,309 @@
+                        fill(-Inf, n), fill(Inf, n),  # variable bounds
+                        fill("", m), fill("", n)      # row and column names
+    )
+
+    # Set some parameters
+    Tulip.set_parameter(linear_model, "OutputLevel", display_level)
+    Tulip.set_parameter(linear_model, "Presolve_Level", 0) # disable presolve
+
+    # Solve the problem
+    Tulip.optimize!(linear_model)
+
+    # Check termination status
+    st = Tulip.get_attribute(linear_model, Tulip.Status())
+
+    # Query objective value
+    li = Tulip.get_attribute(linear_model, Tulip.ObjectiveValue())
+
+    return li
+
+end
+
+function tulip_solve_upper_bound_subproblem(ind::Int, φ_matrix::Array{Float64, 2},
+                                            initial_lower_bound::Vector{Float64},
+                                            initial_upper_bound::Vector{Float64};
+                                            display_level::Int = 0)::Float64
+    @assert length(initial_lower_bound) == length(initial_upper_bound) == size(φ_matrix, 1) "NOMAD.jl error : wrong parameters, φ_matrix dimensions are not consistent with initial lower and upper bound dimensions" 
+    @assert ind in 1:size(φ_matrix, 2) "NOMAD.jl error : wrong parameters, bound index is not consistent with  φ_matrix dimensions"
+
+    # Initialize the model 
+    linear_model = Tulip.Model{Float64}()
+
+    m, n = size(φ_matrix)
+
+    # objective vector
+    c = zeros(n)
+    c[ind] = -1.0
+
+    # define problem 
+    Tulip.load_problem!(linear_model.pbdata,
+                        "LP", # problem name
+                        true, c, 0.0,  # objective
+                        SparseArrays.sparse(φ_matrix), initial_lower_bound, initial_upper_bound, # rows
+                        fill(-Inf, n), fill(Inf, n),  # variable bounds
+                        fill("", m), fill("", n)      # row and column names
+    )
+
+    # Set some parameters
+    Tulip.set_parameter(linear_model, "OutputLevel", display_level)
+    Tulip.set_parameter(linear_model, "Presolve_Level", 0) # disable presolve
+
+    # Solve the problem
+    Tulip.optimize!(linear_model)
+
+    # Check termination status
+    st = Tulip.get_attribute(linear_model, Tulip.Status())
+
+    # Query objective value
+    ui = Tulip.get_attribute(linear_model, Tulip.ObjectiveValue())
+
+    return -ui
+
+end
+
+######################################################
+#              NOMAD linear converters               #
+######################################################
+abstract type AbstractConverter end
+
+function convert_to_x(c::AbstractConverter, z::Vector{Float64})::Vector{Float64} end
+
+function convert_to_z(c::AbstractConverter, x::Vector{Float64})::Vector{Float64} end
+
+function get_nz_dimension(c::AbstractConverter)::Int end
+
+struct SVDConverter <: AbstractConverter
+
+    U::Array{Float64, 2} # U factor
+    S::Vector{Float64} # Singular values
+    Vt::Array{Float64, 2} # V factor
```

Comment:
`Matrix{Float64}` ?
`U` and `Vt` are always dense matrices ?

Reply:
Obviously, some performance could be gained if we analyse more the structure of the matrix. But in blackbox optimization, most of the time, the number of variables is small (max 60), and so the computation cost to compute the SVD or QR factorization is negligeable relatively to the cost of the blackbox.

Reply:
Perfect, thanks for the clarification Ludovic !